packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -388,25 +388,34 @@ export class StripeHelper {
   }
 
   /**
-   * Fetch a customer for the record from Stripe based on email.
+   * Fetch a customer for the record from Stripe based on user id.
    */
   async fetchCustomer(
     uid: string,
-    email: string,
     expand?: string[]
   ): Promise<Stripe.Customer | void> {
-    const customerResponse = await this.stripe.customers
-      .list({ email, expand })
-      .autoPagingToArray({ limit: 20 });
-    if (customerResponse.length === 0) {
+    const accountCustomer = await getAccountCustomerByUid(uid);
+    if (
+      accountCustomer === undefined ||
+      accountCustomer.stripeCustomerId === undefined
```

Comment:
Is this possible?  How does it get to be undefined?

Reply:
@chenba So, the customer account table has stripeCustomerId as a nullable field. This is to "future-proof" the table in the event we want to add additional columns linking a Firefox user to an external id for a different payment provider.

Reply:
So, yes, there should not currently be any cases where the stripeCustomerId is undefined currently, this is just to reflect the possibility that it _might be_ in the future